docs/configuration/index.md

```diff
@@ -890,7 +890,7 @@ These Coordinator static configurations can be defined in the `coordinator/runti
 |`druid.coordinator.startDelay`|The operation of the Coordinator works on the assumption that it has an up-to-date view of the state of the world when it runs, the current ZooKeeper interaction code, however, is written in a way that doesn’t allow the Coordinator to know for a fact that it’s done loading the current state of the world. This delay is a hack to give it enough time to believe that it has all the data.|`PT300S`|
 |`druid.coordinator.load.timeout`|The timeout duration for when the Coordinator assigns a segment to a Historical service.|`PT15M`|
 |`druid.coordinator.kill.pendingSegments.on`|Boolean flag for whether or not the Coordinator clean up old entries in the `pendingSegments` table of metadata store. If set to true, Coordinator will check the created time of most recently complete task. If it doesn't exist, it finds the created time of the earliest running/pending/waiting tasks. Once the created time is found, then for all datasources not in the `killPendingSegmentsSkipList` (see [Dynamic configuration](#dynamic-configuration)), Coordinator will ask the Overlord to clean up the entries 1 day or more older than the found created time in the `pendingSegments` table. This will be done periodically based on `druid.coordinator.period.indexingPeriod` specified.|true|
-|`druid.coordinator.kill.on`|Boolean flag for whether or not the Coordinator should submit kill task for unused segments, that is, permanently delete them from metadata store and deep storage. If set to true, then for all whitelisted datasources (or optionally all), Coordinator will submit tasks periodically based on `period` specified. A whitelist can be set via dynamic configuration `killDataSourceWhitelist` described later.<br /><br />When `druid.coordinator.kill.on` is true, segments are eligible for permanent deletion once their data intervals are older than `druid.coordinator.kill.durationToRetain` relative to the current time. If a segment's data interval is older than this threshold at the time it is marked unused, it is eligible for permanent deletion immediately after being marked unused.|false|
+|`druid.coordinator.kill.on`|Boolean flag to enable the Coordinator to submit a kill task for unused segments and delete them permanently from the metadata store and deep storage. When `druid.coordinator.kill.on` is true: <ul><li>Segments become eligible for permanent deletion once their data intervals are older than `druid.coordinator.kill.durationToRetain`  relative to the current time. If a segment's data interval exceeds this threshold at the time it is marked unused, it becomes eligible for immediate deletion.</li><li>If `killDataSourceWhitelist` specifies data sources, the Coordinator issues kill tasks only for those data sources based on the specified period.</li><li>If `killDataSourceWhitelist` is empty, the Coordinator issues kill tasks for all data sources.</li></ul>|false|
```

Comment:
Simplified the language a bit.
Removed the part about becoming eligible immediately since we now have the concept of a `bufferPeriod`.

```suggestion
|`druid.coordinator.kill.on`|Boolean flag to enable the Coordinator to submit a kill task for unused segments and delete them permanently from the metadata store and deep storage. When `druid.coordinator.kill.on` is true: <ul><li>Segments become eligible for permanent deletion once their data intervals become older than `druid.coordinator.kill.durationToRetain` relative to the current time.</li><li>If `killDataSourceWhitelist` specifies data sources, the Coordinator issues kill tasks only for those data sources.</li><li>If `killDataSourceWhitelist` is empty, the Coordinator issues kill tasks for all data sources.</li></ul>|false|
```

Reply:
Could we also add a brief note on the impact of `bufferPeriod` when determining the intervals or segments to kill (similar to what we have here for `durationToRetain`)? Essentially only unused segments in the metadata store that have a `used_status_last_updated` time earlier than the configured `bufferPeriod` are eligible for deletion.

Reply:
I wonder if we need that since it would already be described in the entry for `bufferPeriod` itself. For that matter, I would suggest even getting rid of the part about `durationToRetain` and `killDataSourceWhitelist`.

Reply:
I agree with removing references to `durationToRetain` and `killDataSourceWhiteList` from the `druid.coordinator.kill.on` description. @kfaraz @abhishekrb19 Do you want to move forward with this change?